### ISSUE TITLE
Show language menu in language selection

### ISSUE THREAD

**Issue by comradekingu** [WeblateOrg/weblate#9532]
### Describe the problem

The top menu items all focus-shift (in language view) because the second entry is "language".

### Describe the solution you'd like

Add "Language" to this menu
![image](https://github.com/WeblateOrg/weblate/assets/13802408/da5faab6-97c4-4c59-9a1c-217594594cfa)

### Describe alternatives you've considered

Also considered adding all the submenus to the top menu, but that is more of an involved approach.

### Screenshots

This is what it usually looks like
![image](https://github.com/WeblateOrg/weblate/assets/13802408/ffa22c67-e21e-4ff8-892b-e3904b48c8d6)

### Additional context

"Administrate" and "Share" could be grayed out?

**Comment by nijel** [comment]
What should the “Language” tab contain when you are already at language level? There are different actions at each level and that's not going to change.

**Comment by yilmazdurmaz** [comment]
Hi @nijel,
I am not sure what is the intention of the OP here, but I have a similar idea for that.

While we are already in Language, having the list of "other" languages as well would prove benefical to switch

**Comment by comradekingu** [comment]
@nijel 
The problem occurs when I am in a specific language.
When I am in https://hosted.weblate.org/projects/openrefine/#languages I can click nb_NO,
but then I am in 
https://hosted.weblate.org/languages/nb_NO/openrefine/
and I can't click "Language" again.

If I was able to click "Language", I would expect it to go back to
https://hosted.weblate.org/projects/openrefine/#languages

That means added functionality, and the first 6 entries in the menu are always in the same place.
I see no downside to it.

**Comment by yilmazdurmaz** [comment]
Yep, we want very similar things here.

the "language" button in a project links to `{host}/projects/{project_name}/#languages`

If I am not missing something, just simply adding the same button when under `{host}/languages/{lang_code}/{project_name}` should do the job.

for me, it is the same button but for a component pages at `{host}/projects/{project_name}/{component_name}/{lang_code}` to have a link to `{host}/projects/{project_name}/{component_name}/#translations`

need I open a new issue, or is it ok to merge these two requests?

**Comment by nijel** [comment]
> That means added functionality, and the first 6 entries in the menu are always in the same place.

These entries are always context-specific. This is not a place for navigation to other contexts.

>  I see no downside to it.

I do see downsides here.

> If I am not missing something, just simply adding the same button when under {host}/languages/{lang_code}/{project_name} should do the job.

I've changed the breadcrumbs link to languages in bbfb24eae6 instead. This is where you should look for navigation to the upper level.

**Comment by comradekingu** [comment]
So, that makes clicking "OpenRefine" in
![image](https://github.com/WeblateOrg/weblate/assets/13802408/de5359ff-89bb-4361-a38b-8f6192d09ef8)
https://hosted.weblate.org/languages/nb_NO/openrefine/

different from doing it in
![image](https://github.com/WeblateOrg/weblate/assets/13802408/2480f253-9c88-482c-b2af-48a9762016c1)
https://hosted.weblate.org/projects/openrefine/#languages
…?

The underlying problem there is that it isn't possible to undo mid-level nesting from the breadcrumbs menu,
which I think would only be possible if there was a button to close each one
Project/components[x]/languages[x]/local language[x]
and
Project/+^/languages/local/language for example, to add the component.


(It already isn't possible to shorten the URL from
https://hosted.weblate.org/projects/openrefine/database/nb_NO/
to
https://hosted.weblate.org/projects/openrefine/nb_NO/
(because that is https://hosted.weblate.org/languages/nb_NO/openrefine/)

However, to get back to the main issue (which is now somewhat worse),
I like buttons to always do the same thing, and are always be in the same place.

This allows quick navigation, as muscle memory kicks in more easily to click something when I know how far away it is.

Knowing it will be somewhere else in some menus is very hard to learn.
(Similar to finding administration or plugins, which confuses some users.)

I go between local language in a component to all strings or a different component in a different language every day,
and it is both hard to do and visually displeasing.

I am sure there is a reason to keep it tidy with a needs-must minimal approach, but here I think usability suffers for what is a very pure technical approach.

**Comment by nijel** [comment]
>  I like buttons to always do the same thing, and are always be in the same place.

Buttons will never be same at all levels. There is no point in attempting that. If you have already chosen language, why it should show *Languages* button?

**Comment by yilmazdurmaz** [comment]
> If you have already chosen language, why it should show _Languages_ button?

Multi-linguals maybe?! The current menu system seems to try breadcrumbs to switch depth in the project, but it is not as intuitive as you may think. the coloring and the size of it is not discriminative enough. Besides, we are used to clicking on bigger buttons with intuitive names.

Instead, having that "language" button on all levels would simply mean "the language list of this depth" or "back to language list of this".
- have it on projects/{project}/{lang_code} and it would go back to the project's language list
- have it on projects/{project}/{component}/{lang_code} and it would go back to the component's language list.

**Comment by comradekingu** [comment]
I am not navigating based on where I am, but rather what I want to do.
Therefore where I happen to be and the particulars of it aren't as interesting to me, just how easy it is to get stuff done.
How things look and function is also important.

Going to another language to translate it (as mentioned) is one thing, but sometimes I only
remember what language a comment is posted in and want to switch to that, for example.

I changed something in a source string, and I want to get to another language to fix all the broken checks, etc.

Or I just want to go to the source language to fix stuff after seeing that it makes more sense to do than submitting a translation beforehand.

Doesn't matter if I am in local language view, or browsing a component, all components, or a component with a selected language, but the logic differs between those.

In Project/languages I can switch back to "Components",
![bilde](https://github.com/WeblateOrg/weblate/assets/13802408/53821b0c-50fe-4f6d-bc67-46403eb9eeb3)

but in Project/languages/selected language I can't
![bilde](https://github.com/WeblateOrg/weblate/assets/13802408/023fa29b-cd40-42a3-937b-c60dc2fa05f7)

In
project/components I can navigate to "languages",
![bilde](https://github.com/WeblateOrg/weblate/assets/13802408/b156ccfd-1e54-4dc4-bafe-4bca6a4c5417)
but in
project/components/selected component I can't.
![bilde](https://github.com/WeblateOrg/weblate/assets/13802408/1ff24426-e215-495d-b6c6-bb5102174624)

If project/components "Language" is already selected, and there is no navigating back to "project/components".

That I can click "project" in the breadcrumbs to view components is not clear to me.
I suppose it is better than having one button for "main components" and one for "language components".

However, if one menu works, I ideally want it to work all the time, and it could do all the things I want to do if just
"language" and "components" is added in respective places.

In my main language "language" and "languages" is the same word, which I suppose doesn't help.

TL;DR The idea that there is some kind of hierarchy (that isn't obvious in the URL) is lost on me,
I just want to get to where I want to be.

**Comment by nijel** [comment]
And then you have other pages with a different set of buttons:

![image](https://github.com/WeblateOrg/weblate/assets/212189/511498b2-4049-4f8e-a036-f8a976966538)

![image](https://github.com/WeblateOrg/weblate/assets/212189/b7fd9032-90ff-4e93-b306-ec8fd875ce4c)

> TL;DR The idea that there is some kind of hierarchy 

Yes, it's project / component / translation hierarchy, with different actions being available in different scopes.

**Comment by nijel** [comment]
Oh, I see you've also created a pull requests https://github.com/WeblateOrg/weblate/pull/9561 https://github.com/WeblateOrg/weblate/pull/9560, even when I repeat here that it's not a good idea. Let me explain it again:

The UI is consistent right now, and what you are trying to introduce breaks that. To go level up, there already is the breadcrumbs line. It has the links you are trying to add. The tabs (or buttons, how you call it) are always actions on the current object. Introducing repeated navigation here makes it less consistent and consumes screen space (which is a scare resource on mobile devices).

**Comment by yilmazdurmaz** [comment]
> To go level up, there already is the breadcrumbs line

Speaking for myself, I don't want to go "level up". I want to go back to the language list of the project/component with less effort.

> consumes screen space (which is a scare resource on mobile devices).

with all those contents on screen (already-crowded), I am not sure this makes any difference, and also modern phones have wide enough screens (even my 7-year-old 5.5'' mid-range phone has FHD screen).

> The tabs (or buttons, how you call it) are always actions on the current object.

I hope you did not hastily rejected my pull requests, because they were already doing that exact thing: 
- `{{ project.get_absolute_url }}#languages` for the "current" project
- `{{ object.component.get_absolute_url }}#translations` for the "current" component

If another button breaks the page, you say, then why have these two links in the breadcrumbs list, for example at the end of it? This one is a better idea than the last change you accepted, as it affects all unsuspecting users now. "space, again" you say? then why not make it a recognizable icon.

I am sure, instead of rejecting, you can come up with a better placement/appearance for these two time-saving additions.

**Comment by orangesunny** [comment]
> > To go level up, there already is the breadcrumbs line
> 
> Speaking for myself, I don't want to go "level up". I want to go back to the language list of the project/component with less effort.

There are solutions for all these situations:
- Level(s) up is solved by breadcrumbs
- Back is solved by the browser. There is a big button for back, as well as keyboard shortcut.
- You can also get wild and duplicate tabs in your browser and go back and up the level respectively :)

> 
> > consumes screen space (which is a scare resource on mobile devices).
> 
> with all those contents on screen (already-crowded), I am not sure this makes any difference, and also modern phones have wide enough screens (even my 7-year-old 5.5'' mid-range phone has FHD screen).

Mobile screens are wide and large these days, but we are not going to fill them up with nonsense buttons. Moreover, some phones are not 9:16, but thinner and taller. The current Weblate UI layout has been tested widely, and we are not going to ruin it with a redundant tab.
> 
> > The tabs (or buttons, how you call it) are always actions on the current object.
> 
> I hope you did not hastily rejected my pull requests, because they were already doing that exact thing:
> 
>     * `{{ project.get_absolute_url }}#languages` for the "current" project
> 
>     * `{{ object.component.get_absolute_url }}#translations` for the "current" component> 

That rejection of PRs was not hasty. It was purely intentional after a proper discussion within the Weblate team.



> 
> If another button breaks the page, you say, then why have these two links in the breadcrumbs list, for example at the end of it? This one is a better idea than the last change you accepted, as it affects all unsuspecting users now. "space, again" you say? then why not make it a recognizable icon.

I am not getting to this _whataboutism_ workout. If you feel that something in Weblate UI is redundant in the current state, feel free to suggest a removal.
> 
> I am sure, instead of rejecting, you can come up with a better placement/appearance for these two time-saving additions.

We are open to the suggestions, but this issue is time-consuming nonsense. Adding redundant UI elements is not viable. You can open a discussion about these ideas in the community and get back here if a better implementation comes up from there: [Ideas discussions](https://github.com/WeblateOrg/weblate/discussions/categories/ideas)

**Comment by yilmazdurmaz** [comment]
> We are open to the suggestions

I am not sure anymore.

**Comment by yilmazdurmaz** [comment]
I had once reported a bug to vscode rep when I was baby steps with it. The bug was on a function almost any programmer is using: copy-paste names of files. It was introduced somewhere 2 years prior (and many version before)

Some guys closed the issue marking it as a duplicate, just because "he thinks" it is a duplicate. it was not. After I told him so and sometime later, he gave me some links. I checked them; one was close but not the same. I stated my reasons, yet, no matter what I say, the issue stayed closed, as with the living bug. It was apparent he took it personally that questioning this decision was questioning his authority. I had no choice but to learn to live with the bug.

1.5 years later, I checked again and saw that, a few months ago, someone had a fix on one of the linked issues. I commented on that commit, and the guy checked my bug report. Just a few weeks later, the bug was fixed. It took almost 4 years to fix a bug, with an extra 2 years just because a grumpy guy refused to acknowledge the problem and kept it away from real solvers.

In all your responses here, I feel the same scent. So, I will drop one last suggestion, and if you stay the same, I will wait until that grumpy (or grumpies) leaves Weblate (however long it takes).

The following code fragments have the language links to go back to the language list of the project/component from the breadcrumbs. They can be placed at the end, maybe in parentheses, and can be made as icons. But they do what we expect them to do: go back to the language list.

<details><summary>weblate/templates/language-project.html</summary>
<p>
It will show `project_name/Languages/lang_code`  and is faster way to get to project language list. There can be an icon instead.

(addition is at line 3)
```html
{% block breadcrumbs %}
<li><a href="{{ project.get_absolute_url }}">{{ project }}</a></li>
<li><a href="{{ project.get_absolute_url }}#languages">{% trans "Languages" %}</a></li>
<li><a href="{% url 'project-language' project=project.slug lang=language.code %}">{{ language }}</a></li>
{% endblock %}
```
</p>
</details> 

<details><summary>weblate/templates/translation.html</summary>
<p>
It will show `project_name/component_name/Languages/lang_code` and is faster way to get to component language list.  There can be an icon instead.

(addition at line 4)
```html
{% block breadcrumbs %}
<li><a href="{{ object.component.project.get_absolute_url }}">{{ object.component.project }}</a></li>
{% include "snippets/component-breadcrumb.html" with object=object.component %}
<li><a href="{{ object.component.get_absolute_url }}#translations">{% trans "Languages" %}</a></li>
{% include "snippets/translation-breadcrumb.html" %}
<a class="pull-right flip" href="{{ object.get_widgets_url }}">
<img src="{% url 'widget-image' project=object.component.project.slug component=object.component.slug widget='svg' color='badge' lang=object.language.code extension='svg' %}?native=1" />
</a>
{% endblock %}
```
</p>
</details> 



<details><summary>weblate/templates/translate.html</summary>
<p>
It will show `project_name/component_name/Languages/lang_code/Translate` and is faster way to get back to component language list while translating a string.  Again, there can be an icon instead.

(addition is at line 6)
```html
{% block breadcrumbs %}
<li><a href="{{ project.get_absolute_url }}">{{ project }}</a></li>
{% if object.component.slug != "-" %}
{% include "snippets/component-breadcrumb.html" with object=object.component %}
{% endif %}
<li><a href="{{ object.component.get_absolute_url }}#translations">{% trans "Languages" %}</a></li>
{% include "snippets/translation-breadcrumb.html" %}
{% if object.component.slug == "-" %}
<li><a href="{{ unit.translation.component.get_absolute_url }}">{{ unit.translation.component.name }}</a></li>
{% endif %}
<li><a href="{{ object.get_translate_url }}">{% trans "Translate" %}</a></li>
<a class="pull-right flip" href="{{ object.get_widgets_url }}">
<img src="{% url 'widget-image' project=project.slug component=object.component.slug widget='svg' color='badge' lang=object.language.code extension='svg' %}?native=1" />
</a>
{% endblock %}
```
</p>
</details>

**Comment by nijel** [comment]
> weblate/templates/language-project.html

I've changed the link to point to languages in https://github.com/WeblateOrg/weblate/commit/bbfb24eae68f4572a08553e31d23faa4724aa0cc 2 days ago (also mentioned in https://github.com/WeblateOrg/weblate/issues/9532#issuecomment-1631989719).

> weblate/templates/translate.html
> weblate/templates/translation.html

`{% include "snippets/component-breadcrumb.html" with object=object.component %}` points to same page as `<li><a href="{{ object.component.get_absolute_url }}#translations">{% trans "Languages" %}</a></li>` you've added.

Can you please clarify how having two neighboring links with the same target and different text improves things?

**Comment by yilmazdurmaz** [comment]
The change you made is not "intuitive". Clicking on the project name "should" take you to project page, not to its languages section. this is for those that I referred "unsuspecting" users. All my suggestions have this "exclusively" named link back to the languages section.

**Comment by comradekingu** [comment]
A further expectancy of what stuff looks like, functions as, and where it will be across menus would not go amiss.

If a button does a thing, what does that button do if it is moved elsewhere?
If a button has a place in menu 1, where is the most obvious place for it in menu 2?
If the breadcrumb menu works one way in one editor, should it be different in editor 2?
If a button is named "information" in one menu, should it be "info" in another?
If the URLs look like
https://hosted.weblate.org/languages/nb_NO/3d-slicer/
and understanding hierarchy is important, then why is it
https://hosted.weblate.org/projects/3d-slicer/3d-slicer/nb_NO/#overview

The indicator stock on a car is located in the same place regardless of whether the driver turns left or right.
It is even there if the car is driving straight on a one-land road with no near exits.
Moving it one way always does the same thing.

If the answer isn't obvious when navigating, it is not part of an intuitive UI.
The overarching logic may or not be simple, but hard to grasp by way of just trying to navigate.

**Comment by nijel** [comment]
I believe most concerns have been addressed with recent changes to bring translating view to all levels (https://github.com/WeblateOrg/weblate/issues/11198) and the navigation changes (https://github.com/WeblateOrg/weblate/issues/15017).

In case there are any concerns left (honestly, I don't remember the whole discussion here and didn't read it fully now), please raise them in a separate issue so that it can be understood without going through lengthy discussion here.

**Comment by yilmazdurmaz** [comment]
Thanks, I guess we will wait a bit, because we can only confirm if it fits after these changes are released.
